pkg/release-controller/release_info.go

```diff
@@ -1003,12 +1003,24 @@ func (r *ExecReleaseInfo) specHash(image string) appsv1.StatefulSetSpec {
 							git config --global user.email test@test.com
 							oc registry login --to /tmp/.docker/config.json
 
-							if [ -x /tmp/git/github.com ]
+							if [ -x /tmp/git/github.com ]; 
 							then
 							  echo "Performing git maintenance..."
-							  for repo in $(find /tmp/git -type d -a -name .git | xargs dirname)
-							  do
-								(echo $repo && cd $repo && git gc && git pull)
+							  for repo in $(find /tmp/git -type d -a -name .git | xargs dirname); do
+								(
+								  echo $repo
+								  cd $repo
+							
+								  # Check for unmerged files
+								  if git status | grep -q "Unmerged paths"; then
+									echo "Repository has unmerged files. Removing the directory and moving to the next repository..."
+									rm -rf $repo  # Remove the conflicted repository
+								  else
+									# No conflicts, perform maintenance
```

Comment:
The root cause behind this issue is that someone rewrote the git history in the impacted repository (which should be a hard NO), but it does happen.  However, this seems like a mighty big hammer for resolving the problem...
Perhaps a:
`git reset --hard HEAD`
or:
`git restore .`
Ideally, I'd like to avoid deleting the repository if we can avoid it.